cmd/builder/internal/builder/config.go

```diff
@@ -34,8 +34,9 @@ var ErrInvalidGoMod = errors.New("invalid gomod specification for module")
 // Config holds the builder's configuration
 type Config struct {
 	Logger          *zap.Logger
-	SkipCompilation bool `mapstructure:"-"`
-	SkipGetModules  bool `mapstructure:"-"`
+	SkipCompilation bool   `mapstructure:"-"`
+	SkipGetModules  bool   `mapstructure:"-"`
+	LDFlags         string `mapstructure:"-"`
```

Comment:
This should be part of "Distribution" and we should not have a flag but be part of the yaml config.

Reply:
Why?

Reply:
Because there we have other flags, and because we want to go with "configuration as a code". Which means to pass the configuration to the builder as a config file merged into the code repository.

Reply:
In my example [here](https://github.com/open-telemetry/opentelemetry-collector/pull/6946/files#r1080692017), you can see that ldflags include git information.  The moment these values are committed and pushed, they are incorrect since it will be a new commit hash with new git tags (our tags include datetimes). If ldflags become a configuration value, my automation would have to add it as part of the build process, and the value won't be able to be committed with the rest of the configuration file.

Reply:
@bogdandrutu, would you be open to moving this value under "Distribution" in the yaml config and allowing it to be set through a flag?